software/libdvi/tmds_encode.S

```diff
@@ -529,3 +533,118 @@ decl_func_x tmds_fullres_encode_loop_16bpp_leftshift_x
 	tmds_fullres_encode_loop_16bpp_leftshift
 decl_func_y tmds_fullres_encode_loop_16bpp_leftshift_y
 	tmds_fullres_encode_loop_16bpp_leftshift
+
+
+// Variant of tmds_fullres_encode_loop_16bpp that reads
+// 8-bit wide pixels packed 4 per word.  The interpolator
+// base is set to a reordered list of TMDS symbols based
+// on a user colour palette.
+
+.macro tmds_palette_encode_loop
```

Comment:
Very nice! I might come through and add some macro goodness here later, along with the other fullres loops.